src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -2379,6 +2379,111 @@ paths:
                 $ref: '#/components/schemas/ErrorModel'
       x-codegen-request-body-name: upgrade
 
+##############################################################################
+## SEARCH API STANDARD ENDPOINTS
+##############################################################################
+  /api/repository/v1/search/{id}/index:
+    post:
+      tags:
+        - search
+        - snapshots
+        - repository
+      description: Create a new search index for a snapshot
+      operationId: createSearchIndex
+      parameters:
+        - name: id
+          in: path
+          description: A UUID to used to identify an object in the repository
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Index to create from SQL
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SearchIndexRequest'
+        required: true
+      responses:
+        200:
+          description: Return index result
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SearchIndexModel'
+        400:
+          description: Bad request - invalid id, badly formed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+        403:
+          description: No permission to see snapshot
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+        404:
+          description: Not found - snapshot id does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/search/query:
+    post:
+      tags:
+        - search
+        - snapshots
+        - repository
+      description: Run a query on the metadata of one or more snapshots
+      operationId: querySearchIndices
+      parameters:
+        - name: offset
```

Comment:
I believe we discussed with Phil to get rid of this, as you don't want to paginate using repeated queries

Reply:
I think that was for the CDA work paginating SQL data. In our case we'll just be using the built-in Elasticsearch pagination.

Reply:
Yeah, that was the CDA project. Not sure what we'll end up doing here, but this seems ok to me for now.